tensorboard/data/server/data_compat.rs

```diff
@@ -118,10 +118,15 @@ impl EventValue {
                 // extending to -/+ DBL_MAX, but can handle buckets of negative width?
                 //
                 // For consistency with the status quo, we replicate this questionable logic here.
+                if hp.bucket.len() != hp.bucket_limit.len() {
+                    return Err(DataLoss);
+                }
                 let num_buckets = hp.bucket.len();
-                let bucket_edges = || hp.bucket_limit.iter().take(num_buckets - 1).copied();
-                let bucket_lefts = iter::once(hp.min).chain(bucket_edges());
-                let bucket_rights = bucket_edges().chain(iter::once(hp.max));
+                // Skip the last `bucket_limit`; it gets replaced by `hp.max`. It's okay to ignore
+                // the edge case at 0 since `.zip()` will stop immediately in that case anyway.
+                let bucket_edges = &hp.bucket_limit[..usize::saturating_sub(num_buckets, 1)];
```

Comment:
This can be `num_buckets.saturating_sub(1)` if you want, but I don’t mind it like this.

Reply:
Ack, thanks for the observation.  I guess I didn't really think about whether `num_buckets` is already `usize`.  Will leave as-is just to avoid re-pushing only for this.